mlflow/pyspark/ml/__init__.py

```diff
@@ -158,7 +158,15 @@ def _traverse_stage(stage):
 
     yield stage
     if isinstance(stage, Pipeline):
-        for stage in stage.getStages():
+        original_sub_stages = stage.getStages()
+
+        try:
+            iter(original_sub_stages)
+        except TypeError:
+            raise ValueError(
```

Comment:
```suggestion
            raise TypeError(
```

nit: can we use `TypeError`?